common/transcript.go

```diff
@@ -60,9 +63,11 @@ func (t *Transcript) DomainSep(label string) {
 // scalar field
 //
 // Note that calling the transcript twice, will yield two different challenges
-func (t *Transcript) ChallengeScalar(label string) fr.Element {
+func (t *Transcript) ChallengeScalar(label []byte) fr.Element {
 	t.DomainSep(label)
 
+	t.state.Write(t.buff.Bytes())
+	t.buff.Reset()
 	// Reverse the endian so we are using little-endian
 	// SetBytes interprets the bytes in Big Endian
 	bytes := t.state.Sum(nil)
```

Comment:
When the client pulls a challenge scalar from the transcript, before calling `t.stsate.Sum(nil)` (L73), we flush the buffer into the sha256-state. So this prevented thousands of sha256-state small writes, with a single big one.

(Thousands in the context of having to append to the transcript a proof with thousands of openings).

From the perspective of the transcript client, all this is transparent since the only thing the client can observe is the pulled challenge. If internally we did short writes or waited "until the end" can't be observed so the behavior is the same.